semgrep/semgrep/semgrep_main.py

```diff
@@ -176,6 +180,19 @@ def main(
         notify_user_of_work(filtered_rules, include, exclude)
 
     respect_git_ignore = not no_git_ignore
+
+    workdir = Path.cwd()
+    ignore_rules_file = f"{workdir}/{IGNORE_FILE_NAME}"
+    if os.path.exists(ignore_rules_file):
+        with open(ignore_rules_file) as f:
+            ignore_patterns = Parser(workdir).parse(f)
+            file_ignore: Optional[FileIgnore] = FileIgnore(
+                base_path=workdir,
+                patterns=ignore_patterns,
+            )
+    else:
+        file_ignore = None
```

Comment:
In the agent, we use a default ignore file if none is provided. Should we duplicate this behavior? @brendongo @nbrahms 

Again see [`get_semgrepignore`](https://github.com/returntocorp/semgrep-action/blob/dfa369e282ff93667611e3abc0b129a1d3aa16d1/src/semgrep_agent/semgrep.py#L122)

Reply:
What do things like eslint do?

Reply:
Also, suggestion to add a `--files` flag to `semgrep`, which dumps the list of scanned files.

Reply:
(cf. https://github.com/BurntSushi/ripgrep/blob/master/GUIDE.md)

Reply:
https://eslint.org/docs/user-guide/configuring/ignoring-code

> In addition to any patterns in the .eslintignore file, ESLint always follows a couple of implicit ignore rules even if the --no-ignore flag is passed. The implicit rules are as follows:
> 
> node_modules/ is ignored.
> dot-files (except for .eslintrc.*), as well as dot-folders and their contents, are ignored.

Eslint ignoring is quite complicated actually, there are some exceptions to those.

Reply:
Resolving since we decided to handle this in a separate PR